src/shared/features/AppletSettings/ReportConfigSetting/ReportConfigSetting.tsx

```diff
@@ -350,12 +363,12 @@ export const ReportConfigSetting = ({
 
   useEffect(() => {
     reset(defaultValues);
-    setSubjectData(setValue, includeRespondentId);
+    setSubjectData(subjectDataProps);
```

Comment:
why not remove this function here and add a dependency `appletData` in useEffect below ?